tests/test_modern.py

```diff
@@ -965,3 +965,141 @@ def test_remove_empty_directories_error(mock_log, system_error):
+        dry_run=True,
+        force=False,
+    )
+    
+    assert result == 0
+    mock_run.assert_called_once()
+    args = mock_run.call_args[0][0]
+    assert 'git' in args
+    assert 'clean' in args
+    assert '-dx' in args
+    assert '-n' in args
+    assert args.count('-e') == 2
+    assert '.idea' in args
+    assert '.vscode' in args
+
+
+@patch('pyclean.modern.subprocess.run')
+def test_run_git_clean_force(mock_run):
+    """
+    Does run_git_clean execute git clean with -f flag when force=True?
+    """
+    mock_run.return_value = Mock(returncode=0, stdout='', stderr='')
+    
+    result = pyclean.modern.run_git_clean(
+        Path('.'),
+        ignore_patterns=['.tox'],
+        dry_run=False,
+        force=True,
+    )
+    
+    assert result == 0
+    mock_run.assert_called_once()
+    args = mock_run.call_args[0][0]
+    assert 'git' in args
+    assert 'clean' in args
+    assert '-dx' in args
+    assert '-f' in args
+    assert '-e' in args
+    assert '.tox' in args
+
+
+@patch('pyclean.modern.subprocess.run')
+def test_run_git_clean_interactive(mock_run):
+    """
+    Does run_git_clean execute git clean with -i flag by default?
+    """
+    mock_run.return_value = Mock(returncode=0, stdout='', stderr='')
+    
+    result = pyclean.modern.run_git_clean(
+        Path('.'),
+        ignore_patterns=[],
+        dry_run=False,
+        force=False,
+    )
+    
+    assert result == 0
+    mock_run.assert_called_once()
+    args = mock_run.call_args[0][0]
+    assert 'git' in args
+    assert 'clean' in args
+    assert '-dx' in args
+    assert '-i' in args
+
+
+@patch('pyclean.modern.subprocess.run')
+def test_run_git_clean_exit_code_128(mock_run):
+    """
+    Does run_git_clean return exit code 128 when not in a git repo?
+    """
+    mock_run.return_value = Mock(
+        returncode=128,
+        stdout='',
+        stderr='fatal: not a git repository',
+    )
+    
+    result = pyclean.modern.run_git_clean(
+        Path('.'),
+        ignore_patterns=[],
+        dry_run=False,
+        force=False,
+    )
+    
+    assert result == 128
+
+
+@skip_if_no_git
+@patch('pyclean.modern.execute_git_clean')
+@patch('pyclean.modern.descend_and_clean')
+def test_pyclean_with_git_clean(mock_descend, mock_git_clean):
+    """
+    Does pyclean call execute_git_clean when --git-clean flag is used?
+    """
+    with ArgvContext('pyclean', '.', '--git-clean'):
+        pyclean.cli.main()
+    
+    assert mock_git_clean.called
+
+
+@skip_if_no_git
+@patch('pyclean.modern.execute_git_clean')
+@patch('pyclean.modern.descend_and_clean')
+def test_pyclean_git_clean_exit_128_warning(mock_descend, mock_git_clean, capsys):
+    """
+    Does pyclean log a warning and continue when git clean returns 128?
+    """
+    with ArgvContext('pyclean', '.', '--git-clean'):
+        pyclean.cli.main()
+    
+    captured = capsys.readouterr()
+    # The warning should be logged
+    assert mock_git_clean.called
```

Comment:
This test doesn't seem to make sense. Maybe it's incomplete?

Also, please don't refer to "exit code 128" as a number. The problem we describe is "no git directory".